guides/14_Content_Security_Policy.md

```diff
@@ -0,0 +1,40 @@
+# Content Security Policy for applications using the Amazon Chime SDK
+
+Modern web applications use [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) to protect users from certain classes of attacks. You can add a security policy by configuring your web server to return the Content-Security-Policy HTTP header or using `<meta>` element to configure a policy. You can add security headers using [Lambda@Edge](https://docs.aws.amazon.com/lambda/latest/dg/lambda-edge.html) and Amazon [CloudFront](https://aws.amazon.com/cloudfront/). For more information, see [Adding HTTP Security Headers Using Lambda@Edge and Amazon CloudFront](https://aws.amazon.com/blogs/networking-and-content-delivery/adding-http-security-headers-using-lambdaedge-and-amazon-cloudfront/).
+
+Include the following in your policy to allow the Amazon Chime SDK access to the meeting, messaging, and Amazon Voice Focus resources:
+
+```
+Content-Security-Policy: connect-src 'self' '*.chime.aws' 'wss://*.chime.aws' '*.amazonaws.com' 'https://*.sdkassets.chime.aws'; script-src 'https://*.sdkassets.chime.aws'; script-src-elem 'https://*.sdkassets.chime.aws'; worker-src 'blob:'; child-src 'blob:'
+```
+
+Note that the `script-src-elem` is not supported in Safari and Firefox. `worker-src` is not supported in Safari.Check browser compatibilities before using the above policy.
+
+To learn more about CSP configuration, continue to the next section.
+
+## Meeting
+
+If you use a [meeting session](https://github.com/aws/amazon-chime-sdk-js#meeting-session) to enable audio, video, and screen sharing in your application, ensure that your policy includes the following:
+
+`connect-src`: `*.chime.aws` `wss://*.chime.aws`
+
+## Messaging
+
+If you use the [Amazon Chime SDK messaging](https://docs.aws.amazon.com/chime/latest/dg/using-the-messaging-sdk.html) with the [Amazon Cognito](https://aws.amazon.com/cognito/), ensure that you add the following:
+
+`connect-src`: `*.amazonaws.com`
+
+## Logging
+
+The Amazon Chime SDK for JavaScript listens to the browser's [security policy violation event](https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent) and logs blocked URI during the meeting session creation. The Amazon Chime SDK for JavaScript outputs the "Security Policy Violation" error messages to the web console if you enable the console logger. The security policy violation event is an experimental technology. See [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/SecurityPolicyViolationEvent#browser_compatibility) for more information before using this in production.
```

Comment:
blocked URIs

outputs "Secu…

Consistent capitalization.

Tell the builder what they need to do — don't just tell them to check browser compatibility! What will go wrong?

Reply:
Will change.